src/main/java/com/jmb/springfactory/service/task/TaskServiceImpl.java

```diff
@@ -0,0 +1,166 @@
+import com.jmb.springfactory.model.bo.QueryTaskObject;
+import com.jmb.springfactory.model.dto.CommentDto;
+import com.jmb.springfactory.model.dto.TaskDto;
+import com.jmb.springfactory.model.dto.WorkLogDto;
+import com.jmb.springfactory.model.entity.Task;
+import com.jmb.springfactory.model.enumeration.TaskStatusEnum;
+import com.jmb.springfactory.service.GenericServiceImpl;
+import com.jmb.springfactory.service.ValidatorService;
+
+import static com.jmb.springfactory.service.UtilsService.notExist;
+import static com.jmb.springfactory.service.UtilsService.addIntoList;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+import com.jmb.springfactory.service.productionorder.ProductionOrderService;
+
+@Service
+public class TaskServiceImpl extends GenericServiceImpl<Task, TaskDto, BusinessObjectBase, Integer> 
+  implements TaskService {
+
+  @Autowired
+  private TaskMySQLService taskMySQLService;
+  
+  @Autowired
+  private ProductionOrderService productionOrderService;
+  
+  @Autowired
+  @Qualifier("taskValidatorService")
+  private ValidatorService taskValidatorService;
+
+  private static final String TASK_INITIAL_STATUS = TaskStatusEnum.OPENED.name();
+
+  @Override
+  public GenericMySQLService<Task, Integer> genericDao() {
+    return taskMySQLService;
+  }
+
+  @Override
+  public Class<? extends Task> getClazz() {
+    return Task.class;
+  }
+
+  @Override
+  public Class<? extends TaskDto> getDtoClazz() {
+    return TaskDto.class;
+  }
+
+  @Override
+  public Class<? extends BusinessObjectBase> getBoClazz() {
+    return BusinessObjectBase.class;
+  }
+  
+  @Override
+  public void update(TaskDto taskDto, Integer idTask) throws ServiceLayerException {
+    taskValidatorService.validateOnUpdate(taskDto);
+    super.update(taskDto, idTask);
+  }
+  
+  @Override
+  public TaskDto save(Integer orderId, TaskDto taskDto) throws ServiceLayerException, NotFoundException, 
+    PersistenceLayerException {
+    
+    addInitialInformation(orderId, taskDto);
+    return super.save(taskDto);
+  }
+  
+  @Override 
+  public void delete(Integer id) {
+    
+    final Consumer<Task> updateAsDeleted = task -> {
+       task.setStatus(TaskStatusEnum.DELETED); 
+       try {
+         taskMySQLService.save(task);
+       } catch (Exception e) {
+         serviceLog.error(String.format("A error has ocurred when update as DELETED the task [%s]", e.getMessage()));
+      }
+    };
+
+    serviceLog.info(String.format("Search task with id %s", id));
+    taskMySQLService.findOne(id).ifPresent(updateAsDeleted);
+  }
+
+  /**
+   * Add essential information to create a task
+   * @param orderId
+   * @param taskDto
+   * @throws NotFoundException
+   * @throws PersistenceLayerException
+   */
+  private void addInitialInformation(Integer orderId, TaskDto taskDto)
+      throws NotFoundException, PersistenceLayerException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)